cmd/containerd-shim-runc-v2/task/service.go

```diff
@@ -700,6 +705,15 @@ func (s *service) processExits() {
 		for _, cp := range cps {
 			s.handleProcessExit(e, cp.Container, cp.Process)
 		}
+
+		//Ensure to send init exited event we skipped before.
+		if skippedProcess.Process != nil && skippedProcess.Process.Pid() != 0 {
+			s.handleProcessExit(skippedEvent, skippedProcess.Container, skippedProcess.Process)
+			var process containerProcess
+			var e runcC.Exit
+			skippedProcess = process
+			skippedEvent = e
+		}
```

Comment:
This change assumes that the pending execs will have been resolved in the above
```go
for _, cp := range cps {
			s.handleProcessExit(e, cp.Container, cp.Process)
		}
```

But I'm not convinced this is necessarily the case, and this isn't how this should be resolved. IIRC, the pending event should be emitted by the closure returned by `preStart`.

See https://github.com/laurazard/containerd/blob/ba07947d9c45fd01dd7080e6320a0c4a920f003e/cmd/containerd-shim-runc-v2/task/service.go#L188-L196

Reply:
There might be a bug in the current code (please point it out), but I'm convinced this isn't the correct way to fix it.